src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,119 @@
 import { Extrapolate } from '../derived/interpolate';
 
+function getVal(config) {
+  'worklet';
+
+  const { type, coef, val, ll, rr, x } = config;
+
+  switch (type) {
+    case Extrapolate.IDENTITY:
+      return x;
+    case Extrapolate.CLAMP:
+      if (coef * val < coef * ll) {
+        return ll;
+      }
+      return rr;
+    case Extrapolate.EXTEND:
+    default:
+      return val;
+  }
+}
+
+function isExtrapolate(value) {
+  'worklet';
+
+  return (
+    value === Extrapolate.EXTEND ||
+    value === Extrapolate.CLAMP ||
+    value === Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  const EXTRAPOLATE_ERROR_MSG = `Reanimated: config object is not valid, please provide valid config, for example:
+    interpolate(value, [inputRange], [outputRange], {
+    extrapolateLeft: 'clamp',
+    extrapolateRight: 'extend',
+  })`;
+
+  const EXTRAPOLATE_ERROR = (
+    extrapolate
+  ) => `Reanimated: not supported value for "${extrapolate}" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+    interpolate(value, [inputRange], [outputRange], {
+      ${extrapolate}: 'clamp',
+  })`;
+
+  if (!type) {
+    throw new Error(
+      `${EXTRAPOLATE_ERROR_MSG} or interpolate(value, [inputRange], [outputRange], 'clamp')`
+    );
+  }
+
+  const hasExtrapolateLeft = Object.prototype.hasOwnProperty.call(
+    type,
+    'extrapolateLeft'
+  );
+  const hasExtrapolateRight = Object.prototype.hasOwnProperty.call(
+    type,
+    'extrapolateRight'
+  );
```

Comment:
Why not do it like here ⬇️?
```suggestion
  const hasExtrapolateLeft = type.hasOwnProperty('extrapolateLeft');
  
  const hasExtrapolateRight = type.hasOwnProperty('extrapolateRight');
```

Reply:
Because of eslint error 😕

Reply:
I assume this is caused by this rule:
https://eslint.org/docs/rules/no-prototype-builtins

We don't use `Object.create` nor we shadow prototype methods, so it was safe to ignore.